src/index.js

```diff
@@ -95,6 +101,11 @@ class CID {
       }
     }
 
+    /**
+     * @type {string}
+     */
+    this.base = base || 'base58btc'
```

Comment:
Is 
```
this.base = base || 'base58btc'
------------^^^^ this
```

the `let base` from the constructor? If yes, would a comment help to clarify that? Perhaps at the `let base` definition, to make clear that it's intentionally there and not just by mistake.

Reply:
Yes, that `base` is the from the `let base`. The confusing thing here is the existing logic in  the constructor... the first case checks if what was passed was a CID, copies the values across directly... e.g [`this.version = cid.version`](https://github.com/multiformats/js-cid/blob/2c06315bca0b3802cbce7f43621b8da474b9e595/src/index.js#L67) and returns early. The reminaing checks mutate the value of the arguments and eventually assign them to the `this` instance being set up the constructor. Neither is good or bad, but having both styles in one constructor is weird.